jetty/src/main/java/com/opentable/logging/jetty/JsonRequestLog.java

```diff
@@ -107,48 +107,48 @@ public void log(final Request request, final Response response)
         final String query = request.getQueryString();
         return HttpV1.builder()
-            .logName("request")
-            .serviceType(CommonLogHolder.getServiceType())
-            .uuid(UUID.randomUUID())
-            .timestamp(Instant.ofEpochMilli(request.getTimeStamp()))
-
-            .method(request.getMethod())
-            .status(response.getStatus())
-            .incoming(true)
-            .url(fullUrl(request))
-            .urlQuerystring(query)
-
-            .duration(TimeUnit.NANOSECONDS.toMicros(
-                Duration.between(
-                    Instant.ofEpochMilli(request.getTimeStamp()),
-                    clock.instant())
-                .toNanos()))
-
-            .bodySize(request.getContentLengthLong())
-            .responseSize(response.getContentCount())
-
-            .acceptLanguage(request.getHeader(HttpHeaders.ACCEPT_LANGUAGE))
-            .anonymousId(request.getHeader(OTHeaders.ANONYMOUS_ID))
-            // mind your r's and rr's
-            .referer(request.getHeader(HttpHeaders.REFERER))
-            .referringHost(request.getHeader(OTHeaders.REFERRING_HOST))
-            .referringService(request.getHeader(OTHeaders.REFERRING_SERVICE))
-            .remoteAddress(request.getRemoteAddr())
-            .requestId(getRequestIdFrom(request, response))
-            .sessionId(request.getHeader(OTHeaders.SESSION_ID))
-            .userAgent(request.getHeader(HttpHeaders.USER_AGENT))
-            .userId(request.getHeader(OTHeaders.USER_ID))
-            .headerOtOriginaluri(request.getHeader(OTHeaders.ORIGINAL_URI))
-
-            .headerOtDomain(request.getHeader(OTHeaders.DOMAIN))
-            .headerHost(request.getHeader(HttpHeaders.HOST))
-            .headerAccept(request.getHeader(HttpHeaders.ACCEPT))
-
-            .headerXForwardedFor(request.getHeader(HttpHeaders.X_FORWARDED_FOR))
-            .headerXForwardedPort(request.getHeader(HttpHeaders.X_FORWARDED_PORT))
-            .headerXForwardedProto(request.getHeader(HttpHeaders.X_FORWARDED_PROTO))
-
-            .build();
+                .logName("request")
+                .serviceType(CommonLogHolder.getServiceType())
+                .uuid(UUID.randomUUID())
+                .timestamp(Instant.ofEpochMilli(request.getTimeStamp()))
+
+                .method(request.getMethod())
+                .status(response.getStatus())
+                .incoming(true)
+                .url(fullUrl(request))
+                .urlQuerystring(query)
+
+                .duration(TimeUnit.NANOSECONDS.toMicros(
+                        Duration.between(
+                                Instant.ofEpochMilli(request.getTimeStamp()),
+                                clock.instant())
+                                .toNanos()))
+
+                .bodySize(request.getContentLengthLong())
+                .responseSize(response.getContentCount())
+                .correlationId(request.getHeader(OTHeaders.CORRELATION_ID))
```

Comment:
should we care if the correlation id is not UUID ?

Reply:
There's no formal definition of it provided to us. So I say no.